source/adios2/engine/sst/SstWriter.cpp

```diff
@@ -130,7 +130,14 @@ SstWriter::SstWriter(IO &io, const std::string &name, const Mode mode,
     }
 }
 
-SstWriter::~SstWriter() { SstStreamDestroy(m_Output); }
+SstWriter::~SstWriter()
+{
+    SstStreamDestroy(m_Output);
+    if (m_BP3Serializer)
```

Comment:
Rather than add an explicit cleanup here, can we just switch `m_BP3Serializer` from a `format::BP3Serializer*` to an `std::unique_ptr<format::BP3Serializer>`?

Reply:
Will update in the next version.